detekt-rules-errorprone/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/UnsafeSealedClassChildSpec.kt

```diff
@@ -0,0 +1,115 @@
+package io.gitlab.arturbosch.detekt.rules.bugs
+
+import io.gitlab.arturbosch.detekt.rules.KotlinCoreEnvironmentTest
+import io.gitlab.arturbosch.detekt.test.assertThat
+import io.gitlab.arturbosch.detekt.test.compileAndLintWithContext
+import org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment
+import org.junit.jupiter.api.Test
+
+@KotlinCoreEnvironmentTest
+class UnsafeSealedClassChildSpec(private val env: KotlinCoreEnvironment) {
+
+    private val subject = UnsafeSealedClassChild()
+
+    @Test
+    fun `report sealed class has normal class as child inside sealed body`() {
+        val code = """
+            sealed class Foo {
+                data class Bar1(val foo: Int): Foo()
+                //The wrong class
+                class Bar2 : Foo()
+                data object Bar3 : Foo()
+            }
+        """.trimIndent()
+
+        val actual = subject.compileAndLintWithContext(env, code)
+        assertThat(actual).hasSize(1)
+    }
+
+    @Test
+    fun `report sealed class has normal class as child outside sealed body`() {
+        val code = """
+            sealed class Foo {
+                data class Bar1(val foo: Int): Foo()
+                data object Bar3 : Foo()
+            }
+            //The wrong class
+            class Bar2 : Foo()
+        """.trimIndent()
+
+        val actual = subject.compileAndLintWithContext(env, code)
+        assertThat(actual).hasSize(1)
+    }
+
+    @Test
+    fun `non report sealed class has object as child outside sealed body`() {
```

Comment:
```suggestion
    fun `does not report sealed class has object as child outside sealed body`() {
```

applies to all the functions